pygwytracing.py

```diff
@@ -701,70 +703,62 @@ def searchgrainstats(df, dfargtosearch, searchvalue1, searchvalue2):
-    # path = '/Users/alicepyne/Dropbox/UCL/DNA MiniCircles/Minicircle Data Edited/Minicircle Manuscript/Nickel'
-    # path = '/Users/alicepyne/Dropbox/UCL/DNA MiniCircles/Paper/Pyne et al/Figure 1/aspectratioanalysis'
-    # path = '/Volumes/GoogleDrive/My Drive/AFM research group /Methods paper/Data/Circular/194 bp'
-    # path = '/Volumes/GoogleDrive/My Drive/AFM research group /Methods paper/Data/Circular'
-    # path = '/Volumes/GoogleDrive/My Drive/AFM research group /Methods paper/Data/NPC'
-    # path = '/Volumes/GoogleDrive/My Drive/AFM research group /Methods paper/Data/Archive/'
-    # path = '/Volumes/GoogleDrive/My Drive/AFM research group /Methods paper/Data/Fortracing'
-    # path = '/Volumes/GoogleDrive/My Drive/AFM research group /Methods paper/Data/Fortracing'
-    path = './'
     
-    # Set sample type here
-    sample_type = 'DNA'
-    # sample_type = 'MAC'
-    # sample_type = 'protein'
-
-
-    # Set file type to look for here
-    fileend = '.spm', '.gwy', '*.[0-9]'
-    filetype = '*.[0-9]'
-    # Set extension to export files as here e.g. '.tiff'
-    extension = '.tiff'
-
-    # Set height scale values to save out
-    minheightscale = -0e-9
-    maxheightscale = 3e-9
-    # maxheightscale = 20e-9
-    # maxheightscale = 50e-9
-    # Set size of the cropped window/2 in pixels
-    # cropwidth = 100e-9
-    # cropwidth = 60e-9
-    cropwidth = 40e-9
-    splitwidth = 2e-6
-    # Set number of bins
-    bins = 25
-
-    # Set the value of different variables, based on the type of the sample. minarea is the minimum size for grain
-    # determination; maxdeviation and mindeviation are the allowable deviations from the median pixel size for
-    # removal of large and small objects
-
-    if sample_type == 'DNA':
-        minarea = 300e-9
-        maxdeviation = 1.3
-        mindeviation = 0.7
-        gaussian = 0.1e-9
-        thresholdingcriteria = 0.75
-
-    elif sample_type == 'protein':
-        minarea = 50e-9
-        maxdeviation = 2.0
-        mindeviation = 0.3
-        gaussian = 0.1e-9
-        thresholdingcriteria = 0.5
-
-    elif sample_type == 'MAC':
-        minarea = 1000e-9
-        maxdeviation = 1.5
-        mindeviation = 0.5
-        gaussian = 0.25e-9
-        thresholdingcriteria = 2.1
+    # Test if config file exists
+    # If it doesn't, create a new config file with default parameters
+    if(not os.path.isfile("config.ini")) :
+        # Copy default_config.ini to config.ini
+        print("No config file found named 'config.ini'")
+        print("Copying default_config.ini to config.ini")
+        copyfile('default_config.ini','config.ini')
+
+    # Read the config file
+    print("Reading config file")
+    config = configparser.ConfigParser()
+    config.read("config.ini")
+    print("Config loaded: ")
+    print(" ")
+
+    # Load and print the current parameters
+
+    # Main section
+    path = config.get("MainSection", "path")
```

Comment:
This relative file path stops the "Tracestats.json" file from being created (on docker at least)
```suggestion
    path = os.path.abspath(config.get("MainSection", "path"))
```
<img width="978" alt="Screenshot 2021-11-22 at 14 30 45" src="https://user-images.githubusercontent.com/91465918/142879728-647982bb-1e24-43a9-9c7d-46cf3d13ee7f.png">

Reply:
Also wanted to say this is super helpful and looks like a great addition to TopoStats, awesome job!

Reply:
I have added your change to the config file branch and resolved the merge conflict.

Reply:
Hopefully I have done it correctly! Also thank you for reviewing it!